### PR TITLE
chore: fix doxygen formatting

### PR DESCRIPTION
**Pull request by vishwa2710**
Because of how our doxygen comments are formatted, they don't appear as tooltips with autocompletion in VSCode, nor when you hover over the functions etc. 

Now, when we write code we can get these tool tips:

Screen Shot 2023-11-30 at 4 34 53 PM

I used the following script to batch update all the files in the repo to follow the proper format.
```
import os
import glob
import re

# Get a list of all header files in the repository
header_files = glob.glob(os.path.join(".", "**/*.hpp"), recursive=True)

# Regular expression to match function definitions and their preceding docstrings
license_pattern = re.compile(r"\/{3} Apache.*", re.MULTILINE)
pattern = re.compile(r".*\/{3}.*", re.MULTILINE)
next_line_pattern = re.compile(r"^\s$")
extra_space_pattern = re.compile(r"\/{3}\s{18}")

for header_file in header_files:
    with open(header_file, "r") as file:
        content = file.readlines()

    # Process each line
    new_content = []
    skip_next_line = False
    for i in range(len(content)):
        if skip_next_line:
            skip_next_line = False
            continue

        line = content[i]

        if license_pattern.match(line):
            new_content.append(line)
            continue

        line = re.sub(r"@brief\s{,18}", "@brief ", line)
        line = re.sub(r"@param\s{,18}", "@param ", line)
        line = re.sub(r"\[in\] ", "", line)
        line = re.sub(r"@return\s{,18}", "@return ", line)
        line = re.sub(r"///\s{8}", "/// ", line)
        line = re.sub(r"@code", "@code{.cpp}", line)
        next_line = content[i + 1] if i + 1 < len(content) else ""

        # Check if the current and next line match the pattern
        if pattern.match(line) and next_line_pattern.match(next_line):
            new_content.append(
                line.rstrip() + "\n"
            )  # Remove trailing spaces and add one newline
            skip_next_line = (
                True  # Skip the next line as it is part of the matched pattern
            )
        else:
            new_content.append(line)

    # Write the modified content back to the file
    with open(header_file, "w") as file:
        file.writelines(new_content)

```

### CODE REVIEW

**Review by Derollez — APPROVED** [review]
Awesome! Let's merge and apply to other folders (sharing the load)?

File: include/OpenSpaceToolkit/Astrodynamics/Dynamics/CentralBodyGravity.hpp [hunk 1]
```diff
@@ -25,90 +25,79 @@ using ostk::physics::time::Instant;
...
     CentralBodyGravity(const Shared<const Celestial>& aCelestial);
 
-    /// @brief                  Constructor
+    /// @brief Constructor
     ///
-    /// @code
-    ///                         const aCelestial = { ... };
-    ///                         const aName = { ... };
-    ///                         CentralBodyGravity centralBodyGravity = { aCelestial, aName };
+    /// @code{.cpp}
```
Comment (Derollez) [inline include/OpenSpaceToolkit/Astrodynamics/Dynamics/CentralBodyGravity.hpp:44]: Should we make the @code systematic? It seems it's applied to come constructor only (can be done separately)

File: include/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Models/Propagated.hpp [hunk 2]
```diff
@@ -37,169 +37,149 @@ using ostk::astro::trajectory::State;
 using ostk::astro::trajectory::orbit::Model;
 using ostk::astro::Dynamics;
 
-/// @brief                      Defines an orbit model that is propagated using numerical propagation
-
+/// @brief     Defines an orbit model that is propagated using numerical propagation
```
Comment (Derollez) [inline include/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Models/Propagated.hpp:40]:
```suggestion
/// @brief     Define an orbit model that is propagated using numerical propagation
```

File: include/OpenSpaceToolkit/Astrodynamics/Trajectory/Sequence.hpp [hunk 3]
```diff
@@ -32,8 +32,7 @@ using ostk::astro::trajectory::State;
 using ostk::astro::flight::system::SatelliteSystem;
 using ostk::astro::dynamics::Thruster;
 
-/// @brief                      Represents a sequence of trajectory segments executed in order.
-
+/// @brief     Represents a sequence of trajectory segments executed in order.
```
Comment (Derollez) [inline include/OpenSpaceToolkit/Astrodynamics/Trajectory/Sequence.hpp:35]:
```suggestion
/// @brief     Represent a sequence of trajectory segments executed in order.
```

File: include/OpenSpaceToolkit/Astrodynamics/Trajectory/State/CoordinatesSubsets/CartesianPosition.hpp [hunk 4]
```diff
@@ -24,31 +24,27 @@ using ostk::physics::time::Instant;
...
+    /// @param aName a name
     CartesianPosition(const String& aName);
 
-    /// @brief              Destructor
-
+    /// @brief Destructor
     ~CartesianPosition();
 
-    /// @brief              Transforms the coordinate subset from one frame to another
+    /// @brief Transforms the coordinate subset from one frame to another
```
Comment (Derollez) [inline include/OpenSpaceToolkit/Astrodynamics/Trajectory/State/CoordinatesSubsets/CartesianPosition.hpp:39]: ### 
```suggestion
    /// @brief Transform the coordinate subset from one frame to another
```

File: include/OpenSpaceToolkit/Astrodynamics/Trajectory/State/CoordinatesSubsets/CartesianVelocity.hpp [hunk 5]
```diff
@@ -27,36 +27,32 @@ using ostk::astro::trajectory::state::CoordinatesSubset;
...
+    /// @param aName a name
     CartesianVelocity(const Shared<const CartesianPosition>& aCartesianPositionSPtr, const String& aName);
 
-    /// @brief              Destructor
-
+    /// @brief Destructor
     ~CartesianVelocity();
 
-    /// @brief              Transforms the coordinate subset from one frame to another
+    /// @brief Transforms the coordinate subset from one frame to another
```
Comment (Derollez) [inline include/OpenSpaceToolkit/Astrodynamics/Trajectory/State/CoordinatesSubsets/CartesianVelocity.hpp:47]:
```suggestion
    /// @brief Transform the coordinate subset from one frame to another
```

File: include/OpenSpaceToolkit/Astrodynamics/Trajectory/State/NumericalSolver.hpp [hunk 6]
```diff
@@ -31,14 +31,12 @@ using ostk::astro::trajectory::State;
 using ostk::astro::RootSolver;
 using MathNumericalSolver = ostk::math::solvers::NumericalSolver;
 
-/// @brief                      Defines an astrodynamics state contextual Numerical Solver. This class inherits from the
-///                             OSTk Mathematics Numerical Solver.
-
+/// @brief     Defines an astrodynamics state contextual Numerical Solver. This class inherits from the
```
Comment (Derollez) [inline include/OpenSpaceToolkit/Astrodynamics/Trajectory/State/NumericalSolver.hpp:34]:
```suggestion
/// @brief     Define an astrodynamics state contextual Numerical Solver. This class inherits from the
```

File: test/Global.test.hpp [hunk 7]
```diff
@@ -1,19 +1,19 @@
...
+{
+namespace global
+{
+
+extern std::string someGlobalString;
+
+}
+}  // namespace test
+}  // namespace astro
+}  // namespace ostk
```
Comment (Derollez) [inline test/Global.test.hpp:19]: Not sure what happened to those 2 files when applying the script 🤔

**Review by Derollez — APPROVED** [review]

File: bindings/python/src/OpenSpaceToolkitAstrodynamicsPy/Utilities/ShiftToString.hpp [hunk 8]
```diff
@@ -2,11 +2,10 @@
...
 #include <sstream>
 
-/// @brief                      Shift to String Template Function
+/// @brief     Shift to String Template Function
 ///
-///                             Template function used for __str__ and __repr__
-///                             methods on classes exposed in python.
-
+///                      Template function used for __str__ and __repr__
+///                      methods on classes exposed in python.
```
[the remark this answers is not in the document]
Comment (vishwa2710) [reply bindings/python/src/OpenSpaceToolkitAstrodynamicsPy/Utilities/ShiftToString.hpp:8]: Probably an edge case in my script. Good catch!

File: bindings/python/src/OpenSpaceToolkitAstrodynamicsPy/Utilities/IterableConverter.hpp [hunk 9]
```diff
@@ -23,8 +23,7 @@ struct ToListConverter
 
 struct IterableConverter
 {
-    /// @brief                  Registers converter from a python iterable type to the provided type
-
+    /// @brief Registers converter from a python iterable type to the provided type
```
[the remark this answers is not in the document]
Comment (vishwa2710) [reply bindings/python/src/OpenSpaceToolkitAstrodynamicsPy/Utilities/IterableConverter.hpp:26]: All the other functions use Registers, also these aren't exposed really so I think it's fine.

File: test/Global.test.hpp (hunk 7 above)
Comment (vishwa2710) [reply test/Global.test.hpp:19]: Yeah me neither.

File: include/OpenSpaceToolkit/Astrodynamics/Dynamics/CentralBodyGravity.hpp (hunk 1 above)
Comment (vishwa2710) [reply include/OpenSpaceToolkit/Astrodynamics/Dynamics/CentralBodyGravity.hpp:44]: Yeah I think adding more documentation should be done in a separate MR.

**Pull request merged**